RecoTauTag/RecoTau/plugins/DeepTauId.cc

```diff
@@ -1185,59 +1730,64 @@ class DeepTauId : public deep_tau::DeepTauBase {
+    get(dnn::neutralIsoPtSum) = getValueNorm(tau_funcs.getNeutralIsoPtSum(tau, tau_index, tau_ref), 57.59f, 155.3f);
     get(dnn::neutralIsoPtSumWeight_over_neutralIsoPtSum) =
-        getValue(tau.tauID("neutralIsoPtSumWeight") / tau.tauID("neutralIsoPtSum"));
+        getValue(tau_funcs.getNeutralIsoPtSumWeight(tau, tau_index, tau_ref) /
+                 tau_funcs.getNeutralIsoPtSum(tau, tau_index, tau_ref));
     get(dnn::neutralIsoPtSumWeightdR03_over_neutralIsoPtSum) =
-        getValue(tau.tauID("neutralIsoPtSumWeightdR03") / tau.tauID("neutralIsoPtSum"));
-    get(dnn::neutralIsoPtSumdR03_over_dR05) = getValue(tau.tauID("neutralIsoPtSumdR03") / tau.tauID("neutralIsoPtSum"));
+        getValue(tau_funcs.getNeutralIsoPtSumdR03Weight(tau, tau_index, tau_ref) /
+                 tau_funcs.getNeutralIsoPtSum(tau, tau_index, tau_ref));
+    get(dnn::neutralIsoPtSumdR03_over_dR05) = getValue(tau_funcs.getNeutralIsoPtSumdR03(tau, tau_index, tau_ref) /
+                                                       tau_funcs.getNeutralIsoPtSum(tau, tau_index, tau_ref));
     get(dnn::photonPtSumOutsideSignalCone) =
-        getValueNorm(tau.tauID("photonPtSumOutsideSignalConedR03"), 1.731f, 6.846f);
-    get(dnn::puCorrPtSum) = getValueNorm(tau.tauID("puCorrPtSum"), 22.38f, 16.34f);
+        getValueNorm(tau_funcs.getPhotonPtSumOutsideSignalCone(tau, tau_index, tau_ref), 1.731f, 6.846f);
+    get(dnn::puCorrPtSum) = getValueNorm(tau_funcs.getPuCorrPtSum(tau, tau_index, tau_ref), 22.38f, 16.34f);
     // The global PCA coordinates were used as inputs during the NN training, but it was decided to disable
     // them for the inference, because modeling of dxy_PCA in MC poorly describes the data, and x and y coordinates
     // in data results outside of the expected 5 std. dev. input validity range. On the other hand,
     // these coordinates are strongly era-dependent. Kept as comment to document what NN expects.
     if (!disable_dxy_pca_) {
-      get(dnn::tau_dxy_pca_x) = getValueNorm(tau.dxy_PCA().x(), -0.0241f, 0.0074f);
-      get(dnn::tau_dxy_pca_y) = getValueNorm(tau.dxy_PCA().y(), 0.0675f, 0.0128f);
-      get(dnn::tau_dxy_pca_z) = getValueNorm(tau.dxy_PCA().z(), 0.7973f, 3.456f);
+      get(dnn::tau_dxy_pca_x) = getValueNorm(tau_funcs.getdxyPCA(tau, tau_index).x(), -0.0241f, 0.0074f);
+      get(dnn::tau_dxy_pca_y) = getValueNorm(tau_funcs.getdxyPCA(tau, tau_index).y(), 0.0675f, 0.0128f);
+      get(dnn::tau_dxy_pca_z) = getValueNorm(tau_funcs.getdxyPCA(tau, tau_index).z(), 0.7973f, 3.456f);
     } else {
       get(dnn::tau_dxy_pca_x) = 0;
       get(dnn::tau_dxy_pca_y) = 0;
       get(dnn::tau_dxy_pca_z) = 0;
     }
-
     const bool tau_dxy_valid =
-        std::isnormal(tau.dxy()) && tau.dxy() > -10 && std::isnormal(tau.dxy_error()) && tau.dxy_error() > 0;
+        std::isnormal(tau_funcs.getdxy(tau, tau_index)) && tau_funcs.getdxy(tau, tau_index) > -10 &&
```

Comment:
it would help to add a single method, like `isAbove(double val, min){return isnormal(val) and val > min;}`
to avoid repeated function calls and also make the code less verbose